src/DatePeriod.php

```diff
@@ -0,0 +1,108 @@
+<?php
+declare(strict_types=1);
+
+namespace EoneoPay\Utils;
+
+use IteratorAggregate;
+
+/**
+ * Extended class that replaces the built in DatePeriod that correctly
+ * handles 31st January + P1M being the 28th of Feburary, not the 3rd
+ * of March.
+ */
+class DatePeriod implements IteratorAggregate
+{
+    /**
+     * @var \DateTime
+     */
+    private $start;
+
+    /**
+     * @var \DateInterval
+     */
+    private $interval;
+
+    /**
+     * @var \DateTime
+     */
+    private $end;
+
+    /**
+     * DatePeriod constructor
+     *
```

Comment:
Full namespaces shouldn't be used within file for classes, these should be imported with use statements - note an issue but should the order be start, end, interval?